torchvision/prototype/transforms/_geometry.py

```diff
@@ -414,75 +414,92 @@ def __init__(
             _check_padding_mode_arg(padding_mode)
 
         self.padding = padding
+        self._parsed_padding = F._geometry._parse_pad_padding(padding) if padding else None  # type: ignore[arg-type]
```

Comment:
This can lead to unexpected behaviour if user updates dynamically `self.padding` : 
```
t = RandomCrop(...)
t.padding = 12
```

Reply:
True. Depending if we want the speed boost of not computing this in every call, we could also make padding a `@property` and recompute only if it is set, e.g.

```py
class RandomCrop:
    def __init__(self, padding):
        self._padding = None
        self._parsed_padding = None
        self.padding = padding

    @property
    def padding(self):
        return self._padding

    @padding.setter
    def padding(self, padding):
        self._padding = padding
        self._parsed_padding = _parse_pad_padding(padding)

```

Reply:
Maybe, `_parse_pad_padding(padding)` is not that slow.
On the other hand we already overriding input arg when we assign them to `self`, e.g `_setup_size` thus we may parse padding in the ctor and directly assign the output to `self.padding` ?

Reply:
> This can lead to unexpected behaviour if user updates dynamically self.padding

This is true not only for this but also for most other transform properties. Perhaps it's worth making read-only (using properties without setters). Shall we have a separate RFC for future handling of this?

Reply:
> On the other hand we already overriding input arg when we assign them to `self`, e.g `_setup_size` thus we may parse padding in the ctor and directly assign the output to `self.padding` ?

I think that is a strong argument. We shouldn't special case `padding` here.

> This is true not only for this but also for most other transform properties. Perhaps it's worth making read-only (using properties without setters). Shall we have a separate RFC for future handling of this?

It would make the implementation more complex in general. IMO we should wait until someone actually requests this and otherwise not go for it. As @vfdev-5 mentioned, we are already overwriting user input for some parameters, so `padding` would just continue to do so.